### PR TITLE
std.process: add missing check for failed malloc()

### PR DESCRIPTION
**Pull request by kubo39**

### CODE REVIEW

File: std/process.d [hunk 1]
```diff
@@ -555,6 +560,7 @@ private Pid spawnProcessImpl(scope const(char[])[] args,
                         close(i);
                     }
                 }
+                free(pfds);
```
Comment (CyberShadow) [inline std/process.d]: Why? The current process is either getting replaced by `execve`, or killed by exiting with an error.

Comment (kubo39) [reply std/process.d]: Ah yes, remove this.

File: std/process.d [hunk 2]
```diff
@@ -510,9 +511,9 @@ private Pid spawnProcessImpl(scope const(char[])[] args,
             if (!(config & Config.inheritFDs))
             {
                 // NOTE: malloc() and getrlimit() are not on the POSIX async
-                // signal safe functions list, but practically this should not
-                // be a problem. Tha Java VM and CPython also use malloc() in
-                // its own implementation.
+                // signal safe functions list, but practically this should
+                // not be a problem. Tha Java VM and CPython also use malloc()
```
Comment (n8sh) [inline std/process.d]: "Tha"

**Comment by kubo39** [comment]
How can I retest buildkite?

**Comment by wilzbach** [comment]
You need to be part of the Buildkite D team (otherwise there's only the old `commit --amend` trick). I just restarted it for you, but if you ping me on Slack with your email, I can add you to Buildkite. Of course, traditional mail instead of Slack works too.

**Comment by kubo39** [comment]
Thanks!

**Comment by wilzbach** [comment]
Restarted it again as Baszile has removed his GitHub account and his packages are no longer available.

**Comment by WalterBright** [comment]
Given that you found this bug in Phobos, their may be more lurking. Please grep and check all instances of `malloc`, `calloc` and `realloc`.

File: std/process.d [hunk 3]
```diff
@@ -98,6 +98,9 @@ version (Windows)
     import std.windows.syserror;
 }
 
+import core.exception : OutOfMemoryError;
+
+import std.exception : enforce;
```
Comment (CyberShadow) [inline std/process.d]: Sorry, why were the local imports moved to a global import? I think that's opposite to current Phobos policy.

Comment (kubo39) [reply std/process.d]: I didn't care about, thank you for remind! Fixed at https://github.com/dlang/phobos/pull/7150

**Pull request merged**